tapestry.js

```diff
@@ -2191,6 +2290,7 @@ function getAspectRatio() {
     }
 }
 
+// return the highest x and y that a node is placed at
 function getNodesDimensions(dataset) {
```

Comment:
I tested your code and it was not working very well in all situations. This function takes into account the actual recorded x and y coordinates, but with the introduction of auto-layout, that doesn't make any sense anymore. We need another method to find out how big the "canvas" for our tapestry should be when it comes to auto-layout - that means we need an algorithm that can estimate how much space the auto-layout needs to fit everything in without making it too big or too small. I have come up with something that you could replace for this function for the time being: 
```
function getNodesDimensions(dataset) {
    var maxPointX = 0;
    var maxPointY = 0;
    
    if (autoLayout) {
        var maxDepth = findMaxDepth(root);
        maxPointX = Math.pow(dataset.nodes.length, 1/maxDepth) * MAX_RADIUS * maxDepth * getAspectRatio();
        maxPointY = Math.pow(dataset.nodes.length, 1/maxDepth) * MAX_RADIUS * maxDepth;
    }
    else {
        for (var index in dataset.nodes) {
            // save max point so we can calculate our tapestry width and height
            if (dataset.nodes[index].fx > maxPointX) {
                maxPointX = dataset.nodes[index].fx;
            }
            if (dataset.nodes[index].fy > maxPointY) {
                maxPointY = dataset.nodes[index].fy;
            }
        }
    }
    
    return {
        'x': maxPointX,
        'y': maxPointY
    };
}
```

Reply:
OK I've altered it to your implementation, I think yours makes a lot more sense yeah. I'm currently testing this alongside some other suggestions you made because it flips around some of the things I was working with